### PR TITLE
Bump Terraform provider version to v2.47.0

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

In this PR:

- Upgraded Terraform provider version to `v2.47.0`
- Bump `crossplane/upjet` to the commit `50919febc5ab`

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested with uptest in the comments below.

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/applications/application.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/groups/member.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/applications/application.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/app/roleassignment.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/applications/certificate.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/directoryroles/role.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/groups/member.yaml"

File: go.mod [hunk 1]
```diff
@@ -4,13 +4,15 @@
 
 module github.com/upbound/provider-azuread
 
-go 1.21
+go 1.21.3
```
Comment (sergenyalcin) [inline go.mod:7]: It seems that the reason for this patch version pin is [here](https://github.com/upbound/terraform-provider-azuread/blob/ce1f46c210208225440f9297be387d2102c9b136/go.mod#L65).

**Review by sergenyalcin — APPROVED** [review]
Thanks @turkenf LGTM! Could you please left a comment about the reason of the failure test before merging the PR.

**Comment by turkenf** [comment]
The creation and import part of the `Member.groups` resource passed successfully, but the uptest failed due to a lack of permissions in the deletion part.

Uptest run: https://github.com/crossplane-contrib/provider-upjet-azuread/actions/runs/8361287843

**Pull request merged**